browser/js/global.js

```diff
@@ -264,6 +264,18 @@ window.app = {
 		// alternatively, maybe someone else (e.g. an integrator) knows more about the situation than we do. In this case, let's
 		// let them override our default
 		hintOnscreenKeyboard: function(hint) {
+			if (global.app
+					&& global.L.Map
+					&& global.L.Map.THIS._docLayer.isCalc()
+					&& hint !== undefined) {
+				var command = {
+					Enable: {
+						type: 'boolean',
+						value: hint
+					}
+				};
+				global.app.socket.sendMessage('uno .uno:MoveKeepInsertMode ' + JSON.stringify(command));
```

Comment:
There is a list of allowed commands for read only mode:
https://github.com/CollaboraOnline/online/blob/61aa037ad9315a95e932b894cca75346c65d14f8/browser/src/control/Toolbar.js#L367

so probably it can be sent via `map.sendUnoCommand('.uno:MoveKeepInsertMode', command)` with whitelisting it

Reply:
thanks, good call!